Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -22,19 +22,23 @@ class Navigator {
     static var tilesURL: URL? = nil
     
     /**
-     Path to the file where history could be stored when `Navigator.dumpHistory(_:)` is called.
+     Path to the directory where history file could be stored when `Navigator.writeHistory(completionHandler:)` is called.
      */
-    static var historyFilePath: URL? = nil
+    static var historyDirectoryURL: URL? = nil
     
     /**
-     Store history to the file stored in `Navigator.historyFilePath` and asynchronously run a callback
-     when dumping finishes.
+     Store history to the directory stored in `Navigator.historyDirectoryURL` and asynchronously run a callback
+     when writing finishes.
      
-     - parameter completion: A block object to be executed when history dumping ends.
+     - parameter completionHandler: A block object to be executed when history dumping ends.
      */
-    func dumpHistory(_ completion: @escaping (String?) -> Void) {
+    func writeHistory(completionHandler: @escaping (URL?) -> Void) {
         historyRecorder.dumpHistory { (path) in
-            completion(path)
+            if let path = path, let historyFileURL = URL(string: path) {
```

Comment:
Does the original string contain a file: URL or a file path?

```suggestion
            if let path = path, let historyFileURL = URL(fileURLWithPath: path) {
```

Reply:
`fileURLWithPath` works fine as well.